### ISSUE TITLE
Custom Template Location not able to be read / can't be found

### ISSUE THREAD

**Issue by ajcampbell1981** [phpDocumentor/phpDocumentor#3160]
I am trying to implement a custom template in my phpdoc.dist.xml file.  However, when I run `php phpDocumentor.phar`, it results in an error that states `The given template ... could not be found or is not readable`.  

## Expected behavior
I would expect that either a) the folder would be able to be found; or b) that it would give a little bit more information as to "where it's looking"  ie: 'Could not find theme in /ROOT_DIR/FOLDER_LOCATION_SEARCHED'.

## Actual behavior
As described above, it is just giving an error that the template could not be found.

## Steps to reproduce the problem

1.  Create a new template in PROJECT_ROOT/phpdocs/templates folder/new_template_name
2.  Add the following line to the phpdoc.dist.xml: `<template name="new_template_name" location="phpdocs/templates/">`
3.  Run the command above `php phpDocumentor.phar`

## Your environment

* Version used: 3.0.0
* PHP version 7
* Operating system and version (e.g. Ubuntu 20.04, Windows 10): Ubuntu 20.04

**Comment by alanef** [comment]
I'm struggling with the same issue. Just seem to able to get it to find a custom template.  Did you resolve?

**Comment by ajcampbell1981** [comment]
> I'm struggling with the same issue. Just seem to able to get it to find a custom template. Did you resolve?

I did not.  Still trying to figure it out.

**Comment by alanef** [comment]
I some how got it working,  but not sure how / what made a difference

I created a folder in my project root  called data/templates   and put the template folder  in there

and set up in my phpdoc.dist.xml in the project root
`<template name="data/templates/fullworks"/>`

any attempt to load from vendor/xyz/etc  didn't seem to work

then I had a fun day deconstructing the 'default' template to strip out and simplify and remove CSS to just what I need - the documentation for the public API methods  and finally wrapping it up in a WordPress shortcode

Still WIP  by this is where I am https://staging.fullworks.net/2022/03/wfea-api-docs/

**Comment by smileBeda** [comment]
Thanks @alanef for that - clearly the DOC is wrong.
It says to pass a _name_ and a _location_ but that won't work
What works is indeed `<template name="relative/path/to/where/your/config/file/is"/>`

@jaapio / @mvriel not sure entirely who is responsible for DOC issues, but this should be addressed, and does not need too much work to fix and avoid people getting frustrated.

On https://docs.phpdoc.org/3.0/guide/references/configuration.html, we should simply either change the "full example" or add a tiny section explaining how to pass a path to the `template` tag.

Thank you!

**Comment by mvriel** [comment]
@smileBeda Actually, everyone can contribute to the documentation. It is in our repo at https://github.com/phpDocumentor/phpDocumentor/tree/master/docs. Once a push is made -or PR is merged- to that folder, our build process will render the documentation (using phpDocumentor itself!) and deploy that to our webservers.

As for the mentioned issue, I haven't fully read this issue but at least wanted to respond to your question about the docs quickly :)

**Comment by mvriel** [comment]
As for this issue, I'd need to review the code. There may be another issue at play. When you use a PHAR file and relative path names in the application, it will attempt to file files inside the PHAR instead of the actual filesystem. A bit of digging may be needed here to check if this is what it going wrong

**Comment by mvriel** [comment]
Ok, this is awkward. I just checked the codebase and the location configuration option seems to be unused? I am not quite sure how that happened, but thank you for reporting this. In order to prepare a fix, I'd need to dive into that part of the code again as filesystem operations are a tad more complex because we need to take various operating systems into account, that you can run phpDocumentor from a PHAR file and that Twig is properly instantiated with the correct order of files.

> then I had a fun day deconstructing the 'default' template to strip out and simplify and remove CSS to just what I need - the documentation for the public API methods and finally wrapping it up in a WordPress shortcode

Nowadays, you do not need a completely custom template if all you want is a couple of overrides. We introduced a mechanism where you can individual parts of the default template using single twig files in a subfolder `.phpdoc/template` in your project. Unfortunately, I haven't been able to document this new feature yet; it is on my TODOlist to do so :)

**Comment by Cliabhach** [comment]
Wait, so _that's_ what https://github.com/phpDocumentor/phpDocumentor/blob/971e252a904c409b379d966b0f43b453c5cc2ec1/data/templates/default/css/custom.css.twig is for? I can just override the theme colors in `.phpdoc/template/css/custom.css.twig` and it'll all work out?

**Comment by mvriel** [comment]
@Cliabhach that is correct, another route may be to add your own https://github.com/phpDocumentor/phpDocumentor/blob/971e252a904c409b379d966b0f43b453c5cc2ec1/data/templates/default/css/variables.css.twig; but that may be a bit trickier since any changes we do may not be included.

There is also an experiment on-going whether we could use Tailwind as a basis to make it even easier; though I do not expect that soon (or ever at all) since there are quite a few kinks to work out there.

**Comment by mvriel** [comment]
I have made a fix for this issue in the PR mentioned above. To complete the PR I need a newer version of phpstan and I am currently checking with @jaapio whether we could upgrade the version in our github action

**Comment by mvriel** [comment]
Ok, it took a bit of work but the issue is resolved on master. Once the build is complete, our Docker image should have the fix immediately and it will be included in the next release of phpDocumentor. If you want to try it out now, every Github Action build on master produces a PHAR file that can immediately be tried out :)

**Comment by gaxweb** [comment]
> Wait, so _that's_ what https://github.com/phpDocumentor/phpDocumentor/blob/971e252a904c409b379d966b0f43b453c5cc2ec1/data/templates/default/css/custom.css.twig is for? I can just override the theme colors in `.phpdoc/template/css/custom.css.twig` and it'll all work out?

Then the documentation is currently wrong, because it says to put that file 

> in the folder .phpdoc/template/css/base.html.twig

which is a folder. But even then it doesn't work for me right now in v3.3.1. The CSS is not included.
